dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -344,12 +327,20 @@ private Object parseFlatField(FieldDef fd, boolean isLittleEndian) throws Unpars
                     dataLen = (int) getInt(4, isLittleEndian);
                 }
 
-                if (dataLen > this.inputStream.available())
+                byte[] bytes = new byte[dataLen];
+
+                int bytesConsumed = 0;
+                while (bytesConsumed < dataLen)
                 {
-                    throw new UnparsableContentException("Data ended prematurely");
+                    int bytesRead = this.inputStream.read(bytes,bytesConsumed,dataLen-bytesConsumed);
```

Comment:
No, we will only get 0 if there is more data to be read but it is not yet available. An example would be dafilesrv locks up for some reason. In which case we would hit the socket timeout and close on our side. This would result in inputStream.read returning -1